### PR TITLE
[18.0][MIG] connector: Migration to 18.0

### PR DESCRIPTION
**Pull request by thienvh332**

### CODE REVIEW

File: connector/components/binder.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        :param unwrap: if True, returns the normal record
+                       else return the binding record
+        :return: a recordset, depending on the value of unwrap,
+                 or an empty recordset if the external_id is not mapped
+        :rtype: recordset
+        """
+        context = self.env.context
+        bindings = self.model.with_context(active_test=False).search(
+            [
+                (self._external_field, "=", tools.ustr(external_id)),
```
Comment (nilshamerlinck) [inline connector/components/binder.py]: `tools.ustr` is deprecated, see `https://github.com/odoo/odoo/pull/173811`

**Review by nguyenminhchien — CHANGES_REQUESTED** [review]
Migrate `test_connector` as well: 1 PR for 2 modules

**Comment by thienvh332** [comment]
Hi @nguyenminhchien ,
I updated it. Thanks

**Review by nguyenminhchien — CHANGES_REQUESTED** [review]
I faced some warnings. Please fix them

```
2024-09-24 09:03:05,829 168545 WARNING v18c_connector odoo.addons.base.models.ir_model: Rule access_test_backend has no group, this is a deprecated feature. Every access-granting rule should specify a group. 
2024-09-24 09:03:05,829 168545 WARNING v18c_connector odoo.addons.base.models.ir_model: Rule access_connector_test_record has no group, this is a deprecated feature. Every access-granting rule should specify a group. 
2024-09-24 09:03:05,829 168545 WARNING v18c_connector odoo.addons.base.models.ir_model: Rule access_connector_test_binding has no group, this is a deprecated feature. Every access-granting rule should specify a group. 
2024-09-24 09:03:05,829 168545 WARNING v18c_connector odoo.addons.base.models.ir_model: Rule access_no_inherits_binding has no group, this is a deprecated feature. Every access-granting rule should specify a group. 
```

**Review by nguyenminhchien — APPROVED** [review]

**Pull request closed without merging**